ch_pipeline/analysis/timing.py

```diff
@@ -45,6 +52,48 @@ def setup(self, tcorr):
             Note that the timing correction must *already* be referenced with respect
             to the times that were used to calibrate if processing stacked data.
         """
+        # Query flag database for date ranges that need timing correction
+        needs_timing_correction_flags = None
+
+        if self.rank == 0:
+            connect_database()
+
+            needs_timing_correction_type = df.DataFlagType.get(
+                name="needs_timing_correction"
+            )
+
+            if needs_timing_correction_type is None:
+                raise RuntimeError(
+                    "Could not find data flag type `needs_timing_correction`"
+                )
+
+            needs_timing_correction_flags = list(
+                df.DataFlag.select().where(
+                    df.DataFlag.type == needs_timing_correction_type
+                )
+            )
+
+            nflags = len(needs_timing_correction_flags)
+            if nflags == 0:
+                raise RuntimeError(
+                    "Database query should return at least one needs_timing_correction flag."
+                )
+
+            self.log.info(
+                f"Queried database and received {nflags} needs_timing_correction flags."
+            )
+
+            # We are extracting the time ranges here from the DataFlag objects
+            # to avoid potential issues with non-0 ranks accidentally doing database queries later on
+            flags = [
+                {"start_time": flag.start_time, "finish_time": flag.finish_time}
+                for flag in needs_timing_correction_flags
+            ]
```

Comment:
I think you can probably remove this in favour of either using the DataFlag instances directly, or just using `.dict()` on the query to fetch dictionaries originally rather than DataFlag instances.

Reply:
Haha. I note that I suggested to not use the DataFlag instances directly. Ooops. In that case maybe just use `.dict()`(`.namedtuple()` would probably also do).

Reply:
> just using .dict() on the query to fetch dictionaries originally rather than DataFlag instances.

I do not understand what you mean by this. =( Because, it is potentially returning a list of instances, so I am not sure where the `dict()` would go. Do you mean

```
flags = [dict(flag) for flag in needs_timing_correction_flags]
    
```

Reply:
Ah, no, sorry, that wasn't clear, especially as I had a typo. I mean use this: http://docs.peewee-orm.com/en/latest/peewee/api.html#BaseQuery.dicts

So you could effectively just do:
```python
flags = list(df.DataFlag.select(df.DataFlag.start_time, df.DataFlag.end_time).where(
    df.DataFlag.type == needs_timing_correction_type
).dicts())
```
and you'd get exactly the same thing, but with peewee doing all the lifting for you, and skipping a bunch of intermediate object construction it no longer needs to do.